### DISCUSSION TITLE
rss/

### DISCUSSION THREAD

**Discussion by giscus** [Capchdo/blog#6] [General]
# rss/

关于 HaoBIT 等。

https://blog.haobit.top/rss/

**Comment by YDX-2147483647** [comment]
最新的进展请参阅[下一条评论](https://github.com/Capchdo/blog/discussions/6#discussioncomment-11113047)。

## rssto.email

[RSS/EMAIL](https://www.rssto.email/) 可能出问题了，现在加新的会 502。 🙏 

[Is the service down? · Issue #2 · aksh-at/rss-to-email](https://github.com/aksh-at/rss-to-email/issues/2)

该项目开源。是否有人了解一点 Clojure？能否帮助自建？

2024年3月6日更新：至于原来的订阅，大部分人坏了，个别人至今仍正常。

## 可能的替代

### ❓ [rss2email/rss2email: Forward RSS feeds to your email address, community maintained](https://github.com/rss2email/rss2email)

Python, 13 years old.

### ❌ ~~[skx/rss2email: Convert RSS feeds to emails](https://github.com/skx/rss2email)~~

Go, 6 years old. Docker available.

> This project began life as a naive port of the python-based [r2e](https://github.com/wking/rss2email) (the above project) utility to golang.

发邮件的问题：

- sendmail 试了一次没成功，还没仔细看原因。

- SMTP 只能 plain auth，而我们的服务商似乎只支持 login auth。（→ [go发送smtp邮件时的踩坑记录——auth login、x509: cannot validate certificate for错误-CSDN博客](https://blog.csdn.net/jiangxuege/article/details/86621433)）另外用 [go-mail/mail: Actively maintained fork of gomail. The best way to send emails in Go.](https://github.com/go-mail/mail) 能正常发邮件。

根本问题：每条通知一封邮件，这样不行，太多了。而且它在设计上很难改，因为[发邮件是`processFeed`的一部分](https://github.com/skx/rss2email/blob/a76100aab701fb09ef5dae0fbb24f2019b63494a/processor/processor.go#L334-L336)。（→ [FR: Newsletter format · Issue #75 · skx/rss2email](https://github.com/skx/rss2email/issues/75)）

### 🤨 [bdevos/rss-to-email: Scheduled RSS to Email with Github Actions](https://github.com/bdevos/rss-to-email/)

Web + GitHub Actions, 11 months old.

挺方便，邮件格式也很漂亮；正在 https://github.com/Capchdo/rss-to-email 调试。

- 可惜因为它按`pubDate`判断是否发过（而非 seen/unseen），而[通知的时间并不单调](https://github.com/YDX-2147483647/bulletin-issues-transferred/discussions/52)。因为发邮件利用 GitHub Actions，和“RSS 转邮件”完全解耦，也许好改。
- 没有 RSS 的正文，不过我们几乎不需要。

### ❓ [WayneGongCN/rss-mailer: The Easy Way to Send RSS Feeds to Email. Avoid information explosion and AI Recommendation Algorithms. Improve the efficiency of information acquisition](https://github.com/WayneGongCN/rss-mailer)

TypeScript, 3 years old.

### 其它

https://github.com/AboutRSS/ALL-about-RSS#rss2mail

**Reply by YDX-2147483647** [reply]
我们下午查了一下。上游项目 rssto.email 加其它源也不行，现在联系不上作者，所以目前什么都做不了。rssto.email 开源，可以自己部署，但发现有些困难，短期（几天内）弄不了。也试了一个类似项目，最后发现它是每条通知一封邮件。

现在有下面两种建议方案。

- 用RSS。
- 用“[i乐学助手](https://apps.apple.com/cn/app/i%E4%B9%90%E5%AD%A6%E5%8A%A9%E6%89%8B/id6468846337)”。这是别的同学开发的软件，功能之一是从 haobit.top 获取通知。

**Reply by YDX-2147483647** [reply]
- https://github.com/bdevos/rss-to-email/ — Web + GitHub Actions
- https://github.com/WayneGongCN/rss-mailer — TypeScript

**Reply by YDX-2147483647** [reply]
现在比较满意 Web + GitHub Actions 的。正在 https://github.com/Capchdo/rss-to-email 调试。

**Reply by YDX-2147483647** [reply]
https://github.com/bdevos/rss-to-email/  也不行，因为它按时间判断是否发过，而[通知的时间不单调](https://github.com/YDX-2147483647/bulletin-issues-transferred/discussions/52)。

**Reply by CZLeader** [reply]
新的解决方案见[下方评论](https://github.com/Capchdo/blog/discussions/6#discussioncomment-11113047)

**Comment by CZLeader** [comment]
现在可以往haomailer@capchdo.com发邮件来实现订阅了。
可以借助以下subscriber来生成订阅申请邮件
https://dev.haobit.top/notice/subscribe

需要注意的是，请按照这个格式写邮件，目前subscriber自动生成的少周x信息
睿信,求是;周一,周二,周三;09:00

放心尝试，格式错误会提醒的

**Comment by fishyy119** [comment]
RSS功能是否还正常？

我使用本地阅读器订阅该RSS但发现好久没有获取到最新内容了，浏览器打开RSS链接发现报文的最新内容停留在了1月14日。

**Reply by YDX-2147483647** [reply]
感谢反馈，应该是之前某次维护服务器时弄错了，我再仔细查查。

**Reply by YDX-2147483647** [reply]
已恢复，具体原因我之后总结到[爆炸记录](https://blog.haobit.top/down/)；不过 rss 只显示最近的若干项，并不能从1月14日续上，也只好作罢了。

其实我前一阵也感觉自己订阅的RSS少通知了，但我看网站上有新通知，误以为是RSS客户端的问题了。

---

已更新：https://github.com/Capchdo/blog/commit/5e204182797676bfd9ef96529602b74bba9b6653
